File: src/main/java/refactown/cleancode/c10classes/ClassesMain.java
package refactown.cleancode.c10classes;

import refactown.cleancode.c10classes.model.*;

public class ClassesMain {

	public static void main(String[] args) {
		Jogador bill = new Jogador("Bill (Primeiro)");
		Jogador bob = new Jogador("Bob (Segundo)");
		Jogada jogada = new JogadaSimples(bill, bob);
		MelhorDeTres melhorDeTres = new MelhorDeTres(jogada, 5);
		melhorDeTres.jogar();
		// Print
		if (melhorDeTres.temVencedor()) {
			System.out.println("Vencedor: " + melhorDeTres.getVencedor().getNome());
		} else {
			System.out.println("Empate");
		}
		System.out.println("Resultados:" + melhorDeTres.getResultados());
	}
}

File: src/main/kotlin/refactown/cleancode/c10classes/ClassesMain.kt
package refactown.cleancode.c10classes

fun main() {
    val bill = Player("Bill (First)")
    val bob = Player("Bob (Second)")
    val match = Match(bill, bob)
    val bestOf = BestOfThree(match, 5)
    bestOf.play()
    // Print Results
    if (bestOf.hasWinner()){
        println("Winner: ${bestOf.winner()}")
    } else {
        println ("Draw")
    }
    println("Results: ${bestOf.results()}")
}

